deployments/sgx_plugin/base/intel-sgx-plugin.yaml

```diff
@@ -32,10 +32,22 @@ spec:
         - name: sgx-provision
           mountPath: /dev/sgx_provision
           readOnly: true
+      - name: nri-sgx-epc
```

Comment:
I can address this in the next rev. The current draft just pulls the NRI bits + the SGX webhook modifications and does not add support for any other deployment options just yet.

I was also checking if NFD can check "NRI is enabled" but that does not seem to be available.

Reply:
So in case an additional node selector would be added, you would like to add it to all deployment options and not just for the one based on the kustomize files (described [here](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md#installation-using-kubectl))?

Should I create an official feature request for that?

Reply:
> This more specific node selector could be an optional `kustomize` patch (or added to the [epc-nfd overlay](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/deployments/sgx_plugin/overlays/epc-nfd/kustomization.yaml)) and perhaps used for the entire `DaemonSet`?

I could add `intel.feature.node.kubernetes.io/sgx` to the `epc-nfd` overlay by default. The "base" daemonSet is kept minimal and I'm likely going to drop the NRI bits to an overlay in the final version to make it user "opt-in".

Reply:
Sorry for the late reply.

I really like the implementation of the NFD label as a kustomize patch as part of the overlay for the intel-sgx-plugin `DaemonSet`.

Also, I think the defining the NFD label as a `nodeSelector` for the new cAdvisor `DaemonSet` is a pretty good idea since you named the directory for the resource files `sgx_epc_metrics`. This clearly points to the use case of providing metrics on EPC consumption.

However, maybe we should add a short comment to the label defined as the `nodeSelector` in the patches? E.g. `# intel.feature.node.kubernetes.io/sgx: 'true' # added to SGX enabled nodes by NFD add-on`. Maybe this is not necessary for the `epc-nfd` overlay, but I think it's beneficial for the cAdvisor resource file, because currently the dependency to the NFD add-on is not specified anywhere.

Moreover, maybe the new cAdvisor component should be mentioned in the [README of the sgx plugin](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md#installation-using-kubectl)? Or you don't want to add it there, because it's maybe only needed until the cAdvisor in the kubelet is supporting the misc metrics?

Reply:
> but I think it's beneficial for the cAdvisor resource file, because currently the dependency to the NFD add-on is not specified anywhere.

I can add the comment my next rebase. I'm not sure the label is going to be needed in the final version because cAdvisor could be used for cluster telemetry in general. Unless of course the desire is to only get the EPC+memory metrics... Let's see.

> Moreover, maybe the new cAdvisor component should be mentioned in the [README of the sgx plugin](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/sgx_plugin/README.md#installation-using-kubectl)? Or you don't want to add it there, because it's maybe only needed until the cAdvisor in the kubelet is supporting the misc metrics?

I will add it later once I have better visibility on the PR readiness (which depends on runc 1.2 release I believe). We are most likely going to need the "standalone" deployment for a while.

Reply:
> Or you don't want to add it there, because it's maybe only needed until the cAdvisor in the kubelet is supporting the misc metrics?

I do not think kubelet will be adding metrics, as there are plans on reducing them:
* https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2371-cri-pod-container-stats/README.md
* https://github.com/kubernetes/kubernetes/issues/68522

Reply:
Right. I had checked with the KEP author on what the best approach for me would be and the suggestion was to start with runc+cAdvisor side of things. Longer run we'd need `misc` to be part of runtimes' metrics which is the future anyway but for now we'd use a standalone cAdvisor.